jenkins/jobs/parallel_e2e_features_test.pipeline

```diff
@@ -0,0 +1,239 @@
+  }
+}
+
+pipeline {
+  agent {
+    label 'metal3-static-workers'
+  }
+  environment {
+    METAL3_CI_USER = "metal3ci"
+    REPO_ORG = "${PROJECT_REPO_ORG}"
+    REPO_NAME = "${PROJECT_REPO_NAME}"
+    UPDATED_REPO = "${ghprbAuthorRepoGitUrl}"
+    REPO_BRANCH = "${ghprbTargetBranch}"
+    UPDATED_BRANCH = "${ghprbActualCommit}"
+    OS_USERNAME = "metal3ci"
+    OS_AUTH_URL = "https://kna1.citycloud.com:5000"
+    OS_USER_DOMAIN_NAME = "CCP_Domain_37137"
+    OS_PROJECT_DOMAIN_NAME = "CCP_Domain_37137"
+    OS_REGION_NAME = "Kna1"
+    OS_PROJECT_NAME = "Default Project 37137"
+    OS_TENANT_NAME = "Default Project 37137"
+    OS_AUTH_VERSION = 3
+    OS_IDENTITY_API_VERSION = 3
+    TEST_EXECUTER_VM_NAME = "${VM_NAME}"
+    BUILD_TAG = "${env.BUILD_TAG}"
+    PR_ID = "${ghprbPullId}"
+    IMAGE_OS = "${IMAGE_OS}"
+    CAPI_VERSION = "${CAPI_VERSION}"
+    CAPIRELEASE = "${CAPIRELEASE}"
+    CAPM3_VERSION = "${CAPM3_VERSION}"
+    CAPM3RELEASE = "${CAPM3RELEASE}"
+    DEFAULT_HOSTS_MEMORY = "${TARGET_NODE_MEMORY}"
+    NUM_NODES = "${NUM_NODES}"
+    TESTS_FOR = "${TESTS_FOR}"
+    SKIP_DELETION = "${SKIP_DELETION}"
+    UPGRADE_TEST = "${UPGRADE_TEST}"
+    EPHEMERAL_TEST = "${EPHEMERAL_TEST}"
+    IRONIC_INSTALL_TYPE = "${params.IRONIC_INSTALL_TYPE}"
+  }
+  stages {
+    stage('SCM') {
+      options {
+        timeout(time: 5, unit: 'MINUTES')
+      }
+      steps {
+        /* Checkout CI Repo */
+        checkout([$class: 'GitSCM',
+          branches: [
+            [name: ci_git_branch]
+          ],
+          doGenerateSubmoduleConfigurations: false,
+          extensions: [
+            [$class: 'WipeWorkspace'],
+            [$class: 'CleanCheckout'],
+            [$class: 'CleanBeforeCheckout']
+          ],
+          submoduleCfg: [],
+          userRemoteConfigs: [
+            [credentialsId: ci_git_credential_id,
+              url: ci_git_url
+            ]
+          ]
+        ])
+        script {
+          CURRENT_START_TIME = System.currentTimeMillis()
+        }
+      }
+    }
+
+    stage('Parallel e2e test Stage') {
+      failFast true
+      parallel {
+        stage('Run e2e pivoting based features') {
+          options {
+            timeout(time: TIMEOUT, unit: 'SECONDS')
+          }
+          environment {
+            GINKGO_FOCUS = "ephemeral"
+          }
+
+          steps {
+            withCredentials([usernamePassword(credentialsId: 'metal3ci_city_cloud_openstack_credentials', usernameVariable: 'OS_USERNAME', passwordVariable: 'OS_PASSWORD')] {
+              [sshUserPrivateKey(credentialsId: 'metal3ci_city_cloud_ssh_keypair', keyFileVariable: 'METAL3_CI_USER_KEY')] {
+                [string(credentialsId: 'metal3-clusterctl-github-token', variable: 'GITHUB_TOKEN')] {
+                  ansiColor('xterm') {
+                    sh "./jenkins/scripts/integration_test.sh"
+                  }
+                }
+              }
+            })
```

Comment:
Sorry I was probably unclear before. I mean we don't need so many brackets. Something like this:

```
  withCredentials([
     usernamePassword(credentialsId: 'metal3ci_city_cloud_openstack_credentials', usernameVariable: 'OS_USERNAME', passwordVariable: 'OS_PASSWORD'),
     sshUserPrivateKey(credentialsId: 'metal3ci_city_cloud_ssh_keypair', keyFileVariable: 'METAL3_CI_USER_KEY'),
     string(credentialsId: 'metal3-clusterctl-github-token', variable: 'GITHUB_TOKEN')
    ] {
      ansiColor('xterm') {
        sh "./jenkins/scripts/integration_test.sh"
      }
  })
```

So only one `[]` where all credentials goes like `[ credential1, credential2]`.